sdk/src/main/java/com/bugsnag/android/AppData.java

```diff
@@ -34,7 +41,12 @@
         this.appContext = appContext;
         this.sessionTracker = sessionTracker;
         appName = getAppName(appContext);
-        packageName = getPackageName(appContext);
+
+        packageName = calculatePackageName(appContext);
+        buildUuid = config.getBuildUUID();
+        durationMs = getDurationMs();
+        foregroundMs = sessionTracker.getDurationInForegroundMs(System.currentTimeMillis());
+        inForeground = sessionTracker.isInForeground();
```

Comment:
is there an issue with pre calculating these fields that they won't be true at the time we read the data?

Reply:
For the purposes of Unity, I don't think so. I believe the intention is that the Unity layer will call something like: `client.generateAppData()` which will instantiate a new object each time.

For Android, I'd anticipate that these values could be slightly different from previous versions, as they should now be calculated at the time of report capture, rather than at the point of serialisation. I think on average it would be different by the order of milliseconds, and worst case, seconds. We can add a general note about this - the values will be more correct IMO.